distributor/cmd/main.go

```diff
@@ -173,8 +174,18 @@ func createKeptnAPI(httpClient *http.Client, env config.EnvConfig) (keptnapi.Kep
 	}
 	if env.PubSubConnectionType() == config.ConnectionTypeHTTP {
 		scheme := "http"
-		parsed, _ := url.Parse(env.KeptnAPIEndpoint)
-		if parsed.Scheme != "" {
+		parsed, err := url.ParseRequestURI(env.KeptnAPIEndpoint)
+		if err != nil {
+			return nil, err
+		}
+
+		if env.KeptnAPIEndpoint != "" && parsed.Scheme == "" {
+			return nil, fmt.Errorf("invalid scheme for keptn endpoint, %s is not http or https", env.KeptnAPIEndpoint)
+		}
+
+		if strings.HasPrefix(parsed.Scheme, "http") {
+			// if no value is assigned to the endpoint than we keep the default scheme
+			// otherwise we override
 			scheme = parsed.Scheme
 		}
```

Comment:
I would expect an error in the `else` case

Reply:
checked for other protocols in the previous branch since an else would also give error for the default empty endpoint